backend/src/main/java/ai/verta/modeldb/versioning/RepositoryDAORdbImpl.java

```diff
@@ -342,9 +343,17 @@ public RepositoryEntity getRepositoryById(Session session, RepositoryIdentificat
         saveBranch(
             session, commitEntity.getCommit_hash(), ModelDBConstants.MASTER_BRANCH, repository);
       }
-      session.getTransaction().commit();
       if (create) {
-        createRoleBindingsForRepository(request, userInfo, repository);
+        try {
+          createRoleBindingsForRepository(request, userInfo, repository);
+          repository.setDeleted(false);
+          session.update(repository);
```

Comment:
we should wrap this is a separate transaction

Reply:
Fixed